### PR TITLE
breaking: remove slot

### LINKED ISSUE #567: [$500] Remove slots
### Describe the feature

Remove the usage of <slot> elements from everything-from compile to runtime

### Additional information

- [X] Would you be willing to help implement this feature?

### PR DESCRIPTION
**Pull request by Aslemammad**
Resolves #567 

- Forking its-fine and making the APIs ref compatible
- removing the marker element and passing the parent/container ref instead 

**Status**

- [ ] Code changes have been tested against prettier, or there are no code changes
- [ ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the codebase
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [ ] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Aslemammad** [comment]
Ahh, it's getting messy because of the v3 rebase!

**Comment by tobySolutions** [comment]
> Ahh, it's getting messy because of the v3 rebase!

Is it cool, now? Can I help?

**Comment by Aslemammad** [comment]
> Is it cool, now? Can I help?

Not yet, but for now let's focus on the implementation, I'll let you know for sure. Thank you so much.

File: packages/react-server/index.ts [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
         if (!el) return;
 
         const currentBlock = blockFactory(props, props?.key);
 
-        globalInfo.mount(currentBlock, el, el.firstChild);
-        patch.current = (newProps: P) => {
-          globalInfo.patch(currentBlock, blockFactory(newProps, newProps.key));
-        };
+        globalInfo.mount(currentBlock, el, el.firstChild),
+          (patch.current = (newProps: P) => {
```
Comment (aidenybai) [inline packages/react-server/index.ts:55]: hmm this doesn't look right

Comment (Aslemammad) [reply packages/react-server/index.ts:55]: oh yea i guess it's a formating thing that happened!

**Comment by tobySolutions** [comment]
> > Is it cool, now? Can I help?
> 
> Not yet, but for now let's focus on the implementation, I'll let you know for sure. Thank you so much.

My pleasure!

**Pull request closed without merging**